utilities/src/main/java/io/syndesis/qe/bdd/validation/FtpValidationSteps.java

```diff
@@ -18,25 +20,47 @@ public FtpValidationSteps() {
         ftpUtils = new FtpUtils(FtpClientManager.getClient());
     }
 
-    @Then("^validate that file \"([^\"]*)\" has been transfered from \"([^\"]*)\" to \"([^\"]*)\" directory$")
-    public void validateThatFileHasBeenTransferedFromToDirectory(String filename, String remoteFromDirectory, String remoteToDirectory) {
+    @Then("^puts \"([^\"]*)\" file with content \"([^\"]*)\" in the FTP directory: \"([^\"]*)\"$")
+    public void putsFileInTheFTPDirectory(String filename, String text, String remoteDirectory) {
+        try {
+            boolean result = ftpUtils.uploadTestFile(filename, text, remoteDirectory);
+            Assertions.assertThat(result).isTrue();
+            boolean isThere = ftpUtils.isThereFile(remoteDirectory + "/" + filename);
+            Assertions.assertThat(isThere).isTrue();
+        } catch (IOException e) {
+            Assertions.fail("Error: " + e);
+        }
+    }
+
+    @And("^validate that file \"([^\"]*)\" has been transfered to \"([^\"]*)\" FTP directory$")
+    public void validateThatFileHasBeenTransferedToDirectory(String filename, String remoteToDirectory) {
         try {
-            boolean isThereFrom = ftpUtils.isThereFile(remoteFromDirectory + "/" + filename);
-            Assertions.assertThat(isThereFrom).isFalse();
             boolean isThereTo = ftpUtils.isThereFile(remoteToDirectory + "/" + filename);
             Assertions.assertThat(isThereTo).isTrue();
         } catch (IOException e) {
             Assertions.fail("Error: " + e);
         }
     }
 
-    @Then("^puts \"([^\"]*)\" file in the FTP \"([^\"]*)\" directory$")
-    public void putsFileInTheFTPDirectory(String filename, String remoteDirectory) {
+
+    @And("^validate that file \"([^\"]*)\" has been transfered from \"([^\"]*)\" FTP directory$")
+    public void validateFileIsNotThere(String filename, String remoteFromDirectory) {
         try {
-            boolean result = ftpUtils.upload(String.format("../%s"), remoteDirectory);
-            Assertions.assertThat(result).isTrue();
-            boolean isThere = ftpUtils.isThereFile(remoteDirectory + "/" + filename);
-            Assertions.assertThat(isThere).isTrue();
+            boolean isThereFrom = ftpUtils.isThereFile(remoteFromDirectory + "/" + filename);
+            Assertions.assertThat(isThereFrom).isFalse();
+        } catch (IOException e) {
+            Assertions.fail("Error: " + e);
+        }
+    }
+
+
+    @Then("^validate that file \"([^\"]*)\" has been transfered from \"([^\"]*)\" to \"([^\"]*)\" FTP directory$")
+    public void validateThatFileHasBeenTransferedFromToDirectory(String filename, String remoteFromDirectory, String remoteToDirectory) {
+        try {
+            boolean isThereFrom = ftpUtils.isThereFile(remoteFromDirectory + "/" + filename);
+            Assertions.assertThat(isThereFrom).isFalse();
```

Comment:
unnecessary variable isThereFrom

Reply:
unncessary in what meaning?

Reply:
the variable is declared only for one-time use, so you don't need to declare it and you can use:

```
assertThat(ftpUtils.isThereFile(remoteFromDirectory + "/" + filename)).isFalse()
```